src/app.js

```diff
@@ -397,17 +397,20 @@ app.route("/dist/*")
  * instead of served straight from disk.
  */
 if (app.locals.gatsbyDevUrl) {
-  // eslint-disable-next-line global-require, import/no-extraneous-dependencies
-  const {createProxyMiddleware} = require("http-proxy-middleware");
 
-  // WebSocket endpoint
-  app.get("/socket.io/", createProxyMiddleware(app.locals.gatsbyDevUrl, {ws: true}));
+  // IIFE because without it, eslint fails thinking await is a reserved word.
+  (async () => {
+    const {createProxyMiddleware} = await import("http-proxy-middleware");
 
-  /* This is the end of the line in gatsbyDevUrl mode, as the proxy middleware
-   * doesn't fallthrough on 404 and even if it did the Gatsby dev server
-   * returns a 200 Ok for missing pages.
-   */
-  app.use(createProxyMiddleware(app.locals.gatsbyDevUrl));
+    // WebSocket endpoint
+    app.get("/socket.io/", createProxyMiddleware(app.locals.gatsbyDevUrl, {ws: true}));
+
+    /* This is the end of the line in gatsbyDevUrl mode, as the proxy middleware
+    * doesn't fallthrough on 404 and even if it did the Gatsby dev server
+    * returns a 200 Ok for missing pages.
+    */
+    app.use(createProxyMiddleware(app.locals.gatsbyDevUrl));
+  })();
```

Comment:
Hmm. I think this is problematic because it's now non-deterministic when the `app.get(…)` and `app.use(…)` handlers get registered, e.g. they may not register until after things later in this file. This matters because registration order is routing order.

Reply:
Oh that's a valid point. How about something like fd145bf357625af4ae0df0883829f6efa7fc43ef?

Are there any other implications of the CJS->ESM switch from synchronous to asynchronous module loading that we should be aware of?

Reply:
Not your fault, but that's fairly ugly IMO. Why can't we use [top-level await](https://nodejs.org/api/esm.html#top-level-await) since these are now ES modules?

Reply:
Oh actually you're right, it runs fine without these hacks. This is just an ESLint error, so I'll look into getting ESLint to behave properly.

Reply:
:pray:

Reply:
Continuing ESLint endeavors here: https://github.com/nextstrain/nextstrain.org/pull/583#discussion_r965396359